apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -420,22 +420,24 @@ type RolloutStrategy struct {
 	// +optional
 	RollingUpdate *RollingUpdateConfig `json:"rollingUpdate,omitempty"`
 
-	// ApplyStrategy describes how to resolve the conflict if the resource to be placed already exists in the target cluster
-	// and is owned by other appliers.
+	// ApplyStrategy describes when and how to apply the hub resource manifest to the target cluster.
 	// +optional
 	ApplyStrategy *ApplyStrategy `json:"applyStrategy,omitempty"`
 }
 
-// ApplyStrategy describes how to resolve the conflict if the resource to be placed already exists in the target cluster
-// and whether it's allowed to be co-owned by other non-fleet appliers.
+// ApplyStrategy describes when and how to apply the resource to the target cluster.
 // Note: If multiple CRPs try to place the same resource with different apply strategy, the later ones will fail with the
 // reason ApplyConflictBetweenPlacements.
 type ApplyStrategy struct {
+	// ConflictResolutionConfig describes what to do when the resource to be placed already exists in the target cluster.
+	// +optional
+	ConflictResolutionConfig *ConflictResolutionConfig `json:"conflictResolutionStrategy,omitempty"`
```

Comment:
With ConflictResolutionConfig + ApplyStrategyType + AllowCoOwership, it is very hard to understand how to use the API.

I assume ApplyStrategyType is about how to resolve field-level conflicts, AllowCoOwership is related to resource-level conflicts (when the resource already exists). Then ConflictResolutionConfig is also about resource-level conflicts.

I suggest to rethink about all the fields as a whole and not try to patch the API by making it even more complicated to think about all the old/new fields in applystrategy and what their impact is.

Reply:
yes, we can organize those fields better if we start from scratch but the API is V1 now so we technically cannot remove/change existing fields.

Reply:
the current thinking is that `ConflictResolutionConfig` describes WHAT to do when the resource already exists on the member cluster.   `ApplyStrategyType` describes HOW to apply if "apply" is the action. The `AllowCoOwnership` technically belongs to ConflictResolutionConfig but we can't move it anymore.

Reply:
AllowCoOwnership is not related to the resolve conflict directly. It only indicates whether it allows co-own even when the object has no conflicts.